tmexp/preprocess.py

```diff
@@ -164,37 +206,51 @@ def preprocess(
         if contents == "":
             remove_file_from_dict(file_path, blob_hash, files_info)
             continue
-        try:
-            ctx = client.parse(
-                filename="", language=lang, contents=contents, timeout=5.0
-            )
-        except Exception:
+        for attempt in range(2):
+            try:
+                start = time.time()
+                ctx = client.parse(
+                    filename="",
+                    language=lang,
+                    contents=contents,
+                    timeout=bblfsh_timeout,
+                )
+                uast = ctx.get_all()
+            except Exception:
+                if time.time() - start > bblfsh_timeout - 0.1 and attempt == 0:
+                    logger.warn("Babelfish timed out, restarting the container ...")
+                    subprocess.call(
+                        ["docker", "restart", bblfsh_container],
+                        stdout=open(os.devnull, "wb"),
```

Comment:
You can probably use https://docs.python.org/3/library/subprocess.html#subprocess.DEVNULL directly

Reply:
yup